microcosm_postgres/factories.py

```diff
@@ -57,14 +63,24 @@ def configure_sqlalchemy_engine(graph):
         database_name,
     )
 
-    return create_engine(
-        uri,
+    connection_args = dict(
         pool_size=graph.config.postgres.pool_size,
         pool_timeout=graph.config.postgres.pool_timeout,
         max_overflow=graph.config.postgres.max_overflow,
         echo=bool(graph.config.postgres.echo),
+        connect_args=dict(
+            sslmode="require" if graph.config.postgres.require_ssl else "prefer"
+        )
     )
 
+    if graph.config.postgres.verify_ssl and graph.config.postgres.require_ssl:
```

Comment:
Note that most dynamic configuration happens via environment variables, which are always strings.

You should use `strtobool` here and change the default values to strings.